beacon-chain/sync/initial-sync/blocks_fetcher.go

```diff
@@ -437,6 +475,35 @@ func (f *blocksFetcher) requestBlocks(
 	return resp, nil
 }
 
+// getPeerLock returns peer lock for a given peer. If lock is not found, it is created.
+func (f *blocksFetcher) getPeerLock(pid peer.ID) *peerLock {
+	f.Lock()
+	defer f.Unlock()
+	if lock, ok := f.peerLocks[pid]; ok {
+		lock.accessed = roughtime.Now()
+		return lock
+	}
+	f.peerLocks[pid] = &peerLock{
+		Mutex:    sync.Mutex{},
+		accessed: roughtime.Now(),
+	}
+	return f.peerLocks[pid]
+}
+
+// removeStalePeerLocks is a cleanup procedure which removes stale locks.
+func (f *blocksFetcher) removeStalePeerLocks(age time.Duration) {
+	f.Lock()
+	defer f.Unlock()
+	for peerID, lock := range f.peerLocks {
+		if time.Since(lock.accessed) >= age {
+			lock.Lock()
+			f.peerLocks[peerID] = nil
```

Comment:
if you are going to delete it anyway is there any point of setting this to nil ?

Reply:
Nice catch - I guess it was "garbage collector worshiping gone wrong", updated!